app.py

```diff
@@ -1,67 +1,129 @@
-
-    # // TODO -- MAKE THIS ACCEPT ANY LAT LONG AS A INPUT
-    # latitude and Longitude and  of the area
-    # NYC KNYC central park weather station is Lat: 40.78°N Lon: 73.97°W
-    # long is not negative here because "TOP" in the request accounts for this.
-
-    # Fetching weather data
-    # Hardcoding to the correct weather station
-    weather_data = requests.get(
-        f'https://api.weather.gov/gridpoints/OKX/33,35/forecast'
+def getContext(lat, lng, default_forecast, timestamp):
+    area_data = requests.get(
+        f'https://api.weather.gov/points/{lat},{lng}'
     ).json()
+    
+    cwa, gridX, gridY = area_data['properties']['cwa'], area_data['properties']['gridX'], area_data['properties']['gridY']
 
-    # Period wise data
-    periodic_data = weather_data['properties']['periods']
-
-    latitude, longitude = weather_data['geometry']['geometries'][0]['coordinates'][
-        0], weather_data['geometry']['geometries'][0]['coordinates'][1]
-
-    # Fetching area data
-    area_data = requests.get(
-        f'https://api.weather.gov/points/{longitude},{latitude}'
+    weather_data = requests.get(
+        f'https://api.weather.gov/gridpoints/{cwa}/{gridX},{gridY}/forecast'
     ).json()
 
-    print(area_data)
+    try:
+        periodic_data = weather_data['properties']['periods']
+    except:
+        context = {
+            'error': True,
+            'title': weather_data['title'],
+            'description': weather_data['detail']
+        }
+        return render_template('index.html', **context)
 
-    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state']
+    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state'] +['',' (Default)'][default_forecast]
 
     context = {
         'timestamp': timestamp,
         'periodic_data': periodic_data,
         'areaDescription': areaDescription
     }
 
-    return render_template('index.html', **context)
+    return context
+
+@app.route('/')
+@app.route('/index')
+@app.route('/<zip_code>')
+def index(zip_code=None):
+    
+    timestamp = datetime.datetime.now()
+
+    if zip_code != None and re.search(r'.*(\d{5}?)$', zip_code):
+        if cache.get(CACHE_1_ZIP) is not None and cache.get(CACHE_1_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_1_ZIP)['context'])
+        elif cache.get(CACHE_2_ZIP) is not None and cache.get(CACHE_2_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_2_ZIP)['context'])
+        elif cache.get(CACHE_3_ZIP)  is not None and cache.get(CACHE_3_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_3_ZIP)['context'])
+        else:
+            lat, lng, default_value = 0, 0, False
```

Comment:
default_value isn't use anywhere? Setting this to a `false` is not needed because it's not referenced again